### ISSUE TITLE
`ERROR 22001 (string_data_right_truncation) value too long for type character varying(255)`

### ISSUE THREAD

**Issue by jamilbk** [CloudFire-LLC/cloudfire#7234]
This seems to happen when an email longer than 255 characters is entered in the OTP request form. It would be nice to emit a user-facing error here as opposed to a Logger error to avoid the false positive GCP alert here.

[GCP Log Link](https://console.cloud.google.com/logs/query;query=resource.type%3D%22gce_instance%22%20AND%20resource.labels.project_id%3D%22firezone-prod%22%20AND%20resource.labels.zone%3D%22us-east1-d%22%20AND%20resource.labels.instance_id%3D%226684212053457774289%22%20%0A%20timestamp%20%3E%3D%20%222024-10-31T15:06:52.000Z%22%20%0A%20timestamp%20%3C%3D%20%222024-11-01T15:01:39.657Z%22%20severity%3E%3DERROR;storageScope=project;pinnedLogId=2024-11-01T10:59:05.004626054Z%2F28mzwzfkhtjlj;cursorTimestamp=2024-11-01T10:59:05.004626054Z;aroundTime=2024-11-01T10:59:05.004626054Z;duration=PT1H?project=firezone-prod)

**Comment by bmanifold** [comment]
I don't believe this is due to the email being longer than 255 characters for a couple of reasons.  

I tried putting in an email longer than 255 in my local instance and no error was thrown.  Along with that, in the stack trace the point in our code where the error is being thrown doesn't involve the email address.

Where this seems to be coming from is the user-agent string being longer than 255.  If you look at the log message in GCP you'll see the user-agent string is longer than 255.

Might need to think about the best way to fix this.

**Comment by thomaseizinger** [comment]
> Where this seems to be coming from is the user-agent string being longer than 255. If you look at the log message in GCP you'll see the user-agent string is longer than 255.

Are we talking connlib's connections? Or is this about browsers?

**Comment by bmanifold** [comment]
@thomaseizinger, this is the browser user-agent string.  Where this is happening is in the login flow while trying to create a token.

Here is the user-agent string for this particular issue:
```
Mozilla/5.0 (Linux; Android 12; V2065 Build/SP1A.210812.003; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/130.0.6723.86 Mobile Safari/537.36  Instagram 354.2.0.47.100 Android (31/12; 300dpi; 720x1475; vivo; V2065; 2026; mt6765; en_US; 655395824)
```

I'm having a difficult time even thinking about how `Instagram` would show up in one of our requests.  I am pretty sure there is a browser in the Instagram app most likely, but how it would be used in a login flow for Firezone has me scratching my head.  Unless of course it got intentionally injected for some reason.

**Comment by thomaseizinger** [comment]
> I'm having a difficult time even thinking about how `Instagram` would show up in one of our requests. I am pretty sure there is a browser in the Instagram app most likely, but how it would be used in a login flow for Firezone has me scratching my head. Unless of course it got intentionally injected for some reason.

Yeah if you click on IG ads, it will open a WebView most of the time. Are we running ads on IG? 😂

**Comment by jamilbk** [comment]
> I don't believe this is due to the email being longer than 255 characters for a couple of reasons.
> 
> I tried putting in an email longer than 255 in my local instance and no error was thrown. Along with that, in the stack trace the point in our code where the error is being thrown doesn't involve the email address.
> 
> Where this seems to be coming from is the user-agent string being longer than 255. If you look at the log message in GCP you'll see the user-agent string is longer than 255.
> 
> Might need to think about the best way to fix this.

Hmm yeah that would explain it. What if we did the following:

- migrate this field to `text`
- enforce a 512 or 1024 character limit in code

**Comment by bmanifold** [comment]
> Hmm yeah that would explain it. What if we did the following:
>
>migrate this field to text
enforce a 512 or 1024 character limit in code

Yeah, had been considering something similar to this and was just trying to read up on "side effects".  It looks like changing from a `varchar(n)` to a `text` may cause a small bit of downtime technically, but I think with the size of the table we have it's probably not going to be noticeable.  

The other thing I was looking at was using a `CHECK` constraint in the DB rather than trying to limit the size in Elixir code.  Thoughts?

**Comment by jamilbk** [comment]
> The other thing I was looking at was using a CHECK constraint in the DB rather than trying to limit the size in Elixir code. Thoughts?

Yeah that seems like a good option! Looks like there's even a helper for it: https://hexdocs.pm/ecto/Ecto.Changeset.html#check_constraint/3
